navigator/frontend/src/applets/contract/ContractComponent.tsx

```diff
@@ -84,6 +84,12 @@ const AgreementText = ({text}: {text: string}) => (
     </span>
 );
 
+const Parties = ({title, parties}: {title: string, parties: string[]}) => (
+    <span><SubHeader><Strong>{title}</Strong></SubHeader>
+        <span>{parties.length > 0 ? parties.join(', ') : 'N/A'}</span>
```

Comment:
Not entirely, empty observers would still be shown if the signatories are there. The current idea is show neither signatories nor observers is there are no signatories. But I can change it to incorporate this suggestion and hide the whole observer field if there are no observers (rather than no signatories).

Reply:
Addressed by https://github.com/digital-asset/daml/pull/1814/commits/e583377357154469e007fd88d1f6f163351f8710